lensless/gradient_descent.py

```diff
@@ -66,40 +66,63 @@ def __init__(self, psf, dtype=None, proj=non_neg, **kwargs):
         proj : :py:class:`function`
             Projection function to apply at each iteration. Default is
             non-negative.
+        initial_est : :py:class:`~numpy.ndarray`
+            Initial estimation for the reconstruction. If none, will create
+            a plain gray estimation.
         """
-
+        self._initial_est = initial_est
         super(GradientDescent, self).__init__(psf, dtype)
         assert callable(proj)
         self._proj = proj
 
     def reset(self):
-
         if self.is_torch:
-
+            if self._initial_est is not None:
+                warnings.warn(
+                    "Error : use of initial estimations is not supported with Pytorch yet"
+                )
             # initial guess, half intensity image
             # for online approach could use last reconstruction
-            psf_flat = self._psf.reshape(-1, self._n_channels)
+            psf_flat = self._psf.reshape(-1, self._psf_shape[3])
             pixel_start = (
                 torch.max(psf_flat, axis=0).values + torch.min(psf_flat, axis=0).values
             ) / 2
             self._image_est = torch.ones_like(self._psf) * pixel_start
 
             # set step size as < 2 / lipschitz
-            Hadj_flat = self._convolver._Hadj.reshape(-1, self._n_channels)
-            H_flat = self._convolver._H.reshape(-1, self._n_channels)
+            Hadj_flat = self._convolver._Hadj.reshape(-1, self._psf_shape[3])
+            H_flat = self._convolver._H.reshape(-1, self._psf_shape[3])
             self._alpha = torch.real(1.8 / torch.max(torch.abs(Hadj_flat * H_flat), axis=0).values)
 
         else:
-
-            # initial guess, half intensity image
-            # for online approach could use last reconstruction
-            psf_flat = self._psf.reshape(-1, self._n_channels)
+            # try to determine the type of the inital estimation, if provided
+            if self._initial_est is not None:
```

Comment:
why have all these checks? why not ensure that `initial_est` is 4D when provided to constructor?